druid-lookups/src/main/java/com/yahoo/maha/maha_druid_lookups/server/lookup/namespace/JDBCExtractionNamespaceCacheFactoryWithLeaderAndFollower.java

```diff
@@ -338,15 +342,19 @@ public Timestamp updateLocalCache(final JDBCExtractionNamespace extractionNamesp
 
     /**
      * Safe KafkaConsumer create/call.
+     * The properties set in here will overwrite your passed in properties,
+     * and correspond to the properties in RocksDB-based lookups.
+     * If a Producer bootstrap writes 25M records, expect a startup backlog of 5 minutes.
      * @param kafkaProperties
      * @return
      */
-    synchronized Consumer<String, byte[]> ensureKafkaConsumer(Properties kafkaProperties) {
-        if(!Objects.nonNull(threadLocalConsumer.get())) {
-            threadLocalConsumer.set(new KafkaConsumer<>(kafkaProperties, new StringDeserializer(), new ByteArrayDeserializer()));
+    synchronized void ensureKafkaConsumer(Properties kafkaProperties, String kafkaTopic) {
+        if(!Objects.nonNull(lookupConsumerMap.get(kafkaTopic))) {
```

Comment:
use concurrentHashMap's atomic operation putIfAbsent() instead of these checks.

Reply:
Thanks!